src/api/index.ts

```diff
@@ -1,26 +1,24 @@
 import { SyncStatePayload, SettingsState } from 'types'
 
-export const requestCategories = () =>
-  new Promise((resolve, reject) => {
-    const data = localStorage.getItem('categories') || '[]'
-
-    if (data) {
-      resolve(JSON.parse(data))
-    } else {
-      reject({ message: 'Something went wrong' })
-    }
-  })
+type IPromiseCallbackFun = (value?: any) => void;
```

Comment:
Rather than using `any` is there a way can we use generics to maintain type safety?

Reply:
It doesn't seem to work at the moment, because `TypeScript` is used when defining `Promise` in `any`. Related links: https://github.com/microsoft/TypeScript/blob/master/lib/lib.es2015.promise.d.ts#L33